lib/location/method_gnss.c

```diff
@@ -623,20 +623,28 @@ static void method_gnss_positioning_work_fn(struct k_work *work)
 	/* Configure GNSS to continuous tracking mode */
 	err = nrf_modem_gnss_fix_interval_set(1);
 
-	uint8_t use_case;
+#if defined(CONFIG_NRF_CLOUD_AGPS_ELEVATION_MASK)
+	err |= nrf_modem_gnss_elevation_threshold_set(CONFIG_NRF_CLOUD_AGPS_ELEVATION_MASK);
+#endif
 
-	switch (gnss_config.accuracy) {
-	case LOCATION_ACCURACY_LOW:
-		use_case = NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START |
-			   NRF_MODEM_GNSS_USE_CASE_LOW_ACCURACY;
-		err |= nrf_modem_gnss_use_case_set(use_case);
-		break;
+	uint8_t use_case = NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START;
+
+	if (gnss_config.accuracy == LOCATION_ACCURACY_LOW) {
+		use_case |= NRF_MODEM_GNSS_USE_CASE_LOW_ACCURACY;
+	}
 
-	case LOCATION_ACCURACY_NORMAL:
-		use_case = NRF_MODEM_GNSS_USE_CASE_MULTIPLE_HOT_START;
+#if defined(CONFIG_NRF_CLOUD_AGPS_DISABLE_SCHEDULED_DOWNLOADS)
+	/* prevent modem from wasting time downloading missing ephemerides */
+	use_case |= NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE;
+	int ret = nrf_modem_gnss_use_case_set(use_case);
+
+	if (ret) {
+		use_case &= ~NRF_MODEM_GNSS_USE_CASE_SCHED_DOWNLOAD_DISABLE;
 		err |= nrf_modem_gnss_use_case_set(use_case);
-		break;
 	}
+#else
+	err |= nrf_modem_gnss_use_case_set(use_case);
+#endif
```

Comment:
Location library uses GNSS in continuous tracking mode, so scheduled downloads do not apply. These changes can be reverted.

Reply:
Is this always going to be the case, @hiltunent ?  Any reason to leave this in case other tracking modes are supported in the future?

Reply:
I don't see why we should want to start using GNSS in another mode so maybe I would also remove it for the reasons of clarity. I don't have a strong opinion about this.

Reply:
I'd personally rather leave it so all the bases are covered (defensive programming), especially in case customers change the tracking mode, but I'll defer to the group.